src/averageAges.js

```diff
@@ -55,7 +65,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = (arguments.length === 2)
+    ? people.filter(child => child.sex === 'm'
+      && people.some(mother => mother.name === child.mother))
+    : people.filter(child => people.some(
+      mother => mother.name === child.mother));
+
+  const MothersAge = children.map(child => ({
```

Comment:
Fix the name of the variable. 
```suggestion
  const mothersAge = children.map(child => ({
```